HANA provider doesn't compile if WITH_GUI is off

```
/home/apkbuild/aports/qgis/src/QGIS-final-3_20_0/src/providers/hana/qgshanaprovider.cpp: In member function 'virtual QList<QgsDataItemProvider*> QgsHanaProviderMetadata::dataItemProviders() const':
/home/apkbuild/aports/qgis/src/QGIS-final-3_20_0/src/providers/hana/qgshanaprovider.cpp:1786:20: error: 'QgsHanaDataItemProvider' does not name a type; did you mean 'QgsDataItemProvider'?
 1786 |   providers << new QgsHanaDataItemProvider;
      |                    ^~~~~~~~~~~~~~~~~~~~~~~
      |                    QgsDataItemProvider
[ 79%] Building CXX object src/providers/hana/CMakeFiles/hanaprovider.dir/qgshanaexpressioncompiler.cpp.o
[ 79%] Building CXX object src/providers/wfs/CMakeFiles/wfsprovider.dir/qgsoapifitemsrequest.cpp.o
make[2]: *** [src/providers/hana/CMakeFiles/hanaprovider_a.dir/build.make:202: src/providers/hana/CMakeFiles/hanaprovider_a.dir/qgshanaprovider.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:4671: src/providers/hana/CMakeFiles/hanaprovider_a.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 79%] Building CXX object src/providers/hana/CMakeFiles/hanaprovider.dir/qgshanafeatureiterator.cpp.o
[ 79%] Building CXX object src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithminterpolatepoint.cpp.o
[ 79%] Building CXX object src/providers/wfs/CMakeFiles/wfsprovider.dir/qgsoapifprovider.cpp.o
[ 79%] Building CXX object src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmintersection.cpp.o
[ 79%] Building CXX object src/providers/hana/CMakeFiles/hanaprovider.dir/qgshanaprimarykeys.cpp.o
[ 79%] Building CXX object src/providers/wfs/CMakeFiles/wfsprovider.dir/qgsoapifutils.cpp.o
[ 79%] Building CXX object src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmjoinbyattribute.cpp.o
[ 79%] Building CXX object src/providers/hana/CMakeFiles/hanaprovider.dir/qgshanaprovider.cpp.o
[ 79%] Building CXX object src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmjoinbylocation.cpp.o
[ 79%] Building CXX object src/providers/hana/CMakeFiles/hanaprovider.dir/qgshanaproviderconnection.cpp.o
[ 79%] Building CXX object src/providers/hana/CMakeFiles/hanaprovider.dir/qgshanaresultset.cpp.o
[ 79%] Building CXX object src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmjoinbynearest.cpp.o
[ 79%] Building CXX object src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmjoinwithlines.cpp.o
[ 79%] Linking CXX shared module ../../../output/lib/qgis/plugins/libwfsprovider.so
/home/apkbuild/aports/qgis/src/QGIS-final-3_20_0/src/providers/hana/qgshanaprovider.cpp: In member function 'virtual QList<QgsDataItemProvider*> QgsHanaProviderMetadata::dataItemProviders() const':
/home/apkbuild/aports/qgis/src/QGIS-final-3_20_0/src/providers/hana/qgshanaprovider.cpp:1786:20: error: 'QgsHanaDataItemProvider' does not name a type; did you mean 'QgsDataItemProvider'?
 1786 |   providers << new QgsHanaDataItemProvider;
      |                    ^~~~~~~~~~~~~~~~~~~~~~~
      |                    QgsDataItemProvider
[ 79%] Built target wfsprovider
[ 79%] Building CXX object src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmkmeansclustering.cpp.o
[ 79%] Building CXX object src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmlinedensity.cpp.o
make[2]: *** [src/providers/hana/CMakeFiles/hanaprovider.dir/build.make:202: src/providers/hana/CMakeFiles/hanaprovider.dir/qgshanaprovider.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 79%] Building CXX object src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmlineintersection.cpp.o
[ 79%] Building CXX object src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmlinesubstring.cpp.o
make[1]: *** [CMakeFiles/Makefile2:4699: src/providers/hana/CMakeFiles/hanaprovider.dir/all] Error 2
```

CC @mrylov

--- a/src/providers/hana/CMakeLists.txt
+++ b/src/providers/hana/CMakeLists.txt
@@ -6,6 +6,7 @@ set (HANA_SRCS
   qgshanaconnection.cpp
   qgshanaconnectionpool.cpp
   qgshanaconnectionstringbuilder.cpp
+  qgshanadataitems.cpp
   qgshanadriver.cpp
   qgshanaexpressioncompiler.cpp
   qgshanafeatureiterator.cpp
@@ -21,7 +22,6 @@ set (HANA_SRCS
 
 if (WITH_GUI)
   set(HANA_SRCS ${HANA_SRCS}
-    qgshanadataitems.cpp
     qgshanadataitemguiprovider.cpp
     qgshananewconnection.cpp
     qgshanaprovidergui.cpp

--- a/src/providers/hana/qgshanadataitemguiprovider.cpp
+++ b/src/providers/hana/qgshanadataitemguiprovider.cpp
@@ -20,6 +20,7 @@
 #include "qgshananewconnection.h"
 #include "qgshanaprovider.h"
 #include "qgshanaproviderconnection.h"
+#include "qgshanasourceselect.h"
 #include "qgshanautils.h"
 #include "qgsnewnamedialog.h"
 
@@ -162,6 +163,16 @@ bool QgsHanaDataItemGuiProvider::handleDrop(
     return connItem->handleDrop( data, schemaItem->name() );
   }
   return false;
+}
+
+QWidget *QgsHanaDataItemGuiProvider::createParamWidget( QgsDataItem *root, QgsDataItemGuiContext )
+{
+  QgsHanaRootItem *rootItem = qobject_cast<QgsHanaRootItem *>( root );
+  if ( rootItem == nullptr )
+    return nullptr;
+  QgsHanaSourceSelect *select = new QgsHanaSourceSelect( nullptr, QgsGuiUtils::ModalDialogFlags, QgsProviderRegistry::WidgetMode::Manager );
+  connect( select, &QgsHanaSourceSelect::connectionsChanged, rootItem, &QgsHanaRootItem::onConnectionsChanged );
+  return select;
 }
 
 void QgsHanaDataItemGuiProvider::newConnection( QgsDataItem *item )

--- a/src/providers/hana/qgshanadataitemguiprovider.h
+++ b/src/providers/hana/qgshanadataitemguiprovider.h
@@ -37,6 +37,8 @@ class QgsHanaDataItemGuiProvider : public QObject, public QgsDataItemGuiProvider
     bool acceptDrop( QgsDataItem *item, QgsDataItemGuiContext context ) override;
     bool handleDrop( QgsDataItem *item, QgsDataItemGuiContext context, const QMimeData *data, Qt::DropAction action ) override;
 
+    QWidget *createParamWidget( QgsDataItem *root, QgsDataItemGuiContext ) override;
+
   private:
     static void newConnection( QgsDataItem *item );
     static void editConnection( QgsDataItem *item );

--- a/src/providers/hana/qgshanadataitems.cpp
+++ b/src/providers/hana/qgshanadataitems.cpp
@@ -19,15 +19,12 @@
 #include "qgshanaconnection.h"
 #include "qgshanaconnectionpool.h"
 #include "qgshanaexception.h"
-#include "qgshananewconnection.h"
 #include "qgshanadataitems.h"
 #include "qgshanasettings.h"
-#include "qgshanasourceselect.h"
 #include "qgshanautils.h"
 #include "qgslogger.h"
 #include "qgsmessageoutput.h"
 #include "qgsmimedatautils.h"
-#include "qgsnewnamedialog.h"
 #include "qgsvectorlayer.h"
 #include "qgsvectorlayerexporter.h"
 #include "qgsfieldsitem.h"
@@ -414,15 +411,6 @@ QVector<QgsDataItem *> QgsHanaRootItem::createChildren()
     connections << new QgsHanaConnectionItem( this, connName, mPath + '/' + connName );
   }
   return connections;
-}
-
-QWidget *QgsHanaRootItem::paramWidget()
-{
-  QgsHanaSourceSelect *select = new QgsHanaSourceSelect( nullptr, Qt::WindowFlags(),
-      QgsProviderRegistry::WidgetMode::Manager );
-  connect( select, &QgsHanaSourceSelect::connectionsChanged, this,
-           &QgsHanaRootItem::onConnectionsChanged );
-  return select;
 }
 
 void QgsHanaRootItem::onConnectionsChanged()

--- a/src/providers/hana/qgshanadataitems.h
+++ b/src/providers/hana/qgshanadataitems.h
@@ -40,8 +40,6 @@ class QgsHanaRootItem : public QgsConnectionsRootItem
 
     QVariant sortKey() const override { return 3; }
 
-    QWidget *paramWidget() override;
-
   public slots:
     void onConnectionsChanged();
 };

--- a/src/providers/hana/qgshanaprovider.cpp
+++ b/src/providers/hana/qgshanaprovider.cpp
@@ -32,11 +32,7 @@
 #include "qgshanaresultset.h"
 #include "qgshanacrsutils.h"
 #include "qgshanautils.h"
-#ifdef HAVE_GUI
 #include "qgshanadataitems.h"
-#include "qgshanasourceselect.h"
-#include "qgssourceselectprovider.h"
-#endif
 #include "qgslogger.h"
 #include "qgsmessagelog.h"
 #include "qgsrectangle.h"

--- a/src/providers/hana/qgshanaprovider.h
+++ b/src/providers/hana/qgshanaprovider.h
@@ -21,9 +21,6 @@
 #include "qgsprovidermetadata.h"
 #include "qgshanaconnection.h"
 #include "qgshanaprimarykeys.h"
-#ifdef HAVE_GUI
-#include "qgsproviderguimetadata.h"
-#endif
 #include "qgsvectordataprovider.h"
 
 #include <QVersionNumber>

